docs/architecture-decisions/flag-type.md

```diff
@@ -12,45 +12,44 @@ updated: 2025-08-14
+* If the `flagdType`field is present, `flagd` must validate that all variant values of the flag conform to this type during initialization.
 * Type mismatches found during initialization must be reported as errors.
-* The `Resolve<T>` and `ResolveAll` methods must use the `type` field for validation when it is available.
+* The `Resolve<T>` and `ResolveAll` methods must use the `flagdType`field for validation when it is available.
 * The implementation must be consistent with the OpenFeature specification and the flag manifest schema.
 
 
 ## Considered Options
 
 * **Consistent `defaultVariant` Validation:** Align the behavior of `Resolve<T>` with `ResolveAll` by making `Resolve<T>` validate the evaluated variant against the type of the `defaultVariant`.
-* **API Extension with Explicit Flag Type:** Introduce an optional `type` property to the flag definition to serve as the authoritative source for type validation.
+* **API Extension with Explicit Flag Type:** Introduce an optional `flagdType`property to the flag definition to serve as the authoritative source for type validation.
 
 
 ## Proposal
 
-This proposal is to extend the flag definition with an optional `type` property. This approach is chosen over simply aligning the `Resolve<T>` and `ResolveAll` validation because it addresses the root cause of the type inconsistency and provides a more robust, long-term solution.
+This proposal is to extend the flag definition with an optional `flagdType`property. This approach is chosen over simply aligning the `Resolve<T>` and `ResolveAll` validation because it addresses the root cause of the type inconsistency and provides a more robust, long-term solution.
 
-By introducing an explicit `type` field, it establishes a single source of truth for the flag's type, independent of its variants. This allows for early and consistent type validation during flag definition parsing, preventing type-related errors at runtime.
+By introducing an explicit `flagdType`field, it establishes a single source of truth for the flag's type, independent of its variants. This allows for early and consistent type validation during flag definition parsing, preventing type-related errors at runtime.
 
-The new `type` field will be optional to maintain backward compatibility with existing flag configurations. If the field is omitted, `flagd` will treat the flag as having `object`, and no type validation will be performed against the `defaultVariant`. When the `type` field is present, `flagd` will enforce that all variants of the flag conform to the specified type.
+The new `flagdType`field will be optional to maintain backward compatibility with existing flag configurations. If the field is omitted, `flagd` will treat the flag as having `object`, and no type validation will be performed against the `defaultVariant`. When the `flagdType`field is present, `flagd` will enforce that all variants of the flag conform to the specified type.
```

Comment:
Could you add more details about how the validation would work? 

I'm particularly interested in the cases that the literal values set to the variants do not fit with the flag type.  Currently flagd seems to just cast the value to the type, **with** precision lost (this is perhaps more related to the JSONLogic engine impl). 

CC @toddbaert